### PR TITLE
refactor: Organised Extremeware and XOS tests #5850

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Fixes: #5850 

We have zero devices in stats detected as xos yes all the extremeware devices have XOS in the sysDescr. I've shifted discovery around to pick them both up (well I'm guessing at extremeware as  we have no data for this). I actually think we should remove extremeware but left in as a fail safe.

It does mean I've had to blank out the extremeware test.

Please also merge: https://github.com/librenms/librenms.github.io/pull/82 - I've put down monday the 6th of March as when that notification is to be merged so if we don't get to an agreement by then please don't merge it.

### CODE REVIEW

File: includes/definitions/xos.yaml [hunk 1]
```diff
@@ -2,12 +2,11 @@ os: xos
...
-group: extremeware
 icon: extreme
 over:
     - { graph: device_bits, text: 'Device Traffic' }
     - { graph: device_processor, text: 'CPU Usage' }
     - { graph: device_mempool, text: 'Memory Usage' }
 discovery:
     - sysDescr:
-        - XOS
+      - XOS
```
Comment (murrant) [inline includes/definitions/xos.yaml]: Can we use ExtremeXOS instead?

File: includes/definitions/extremeware.yaml [hunk 2]
```diff
@@ -8,5 +8,6 @@ over:
     - { graph: device_processor, text: 'CPU Usage' }
     - { graph: device_mempool, text: 'Memory Usage' }
 discovery:
-    - sysObjectId:
-        - .1.3.6.1.4.1.1916.2
+    -
+      sysObjectId: .1.3.6.1.4.1.1916.2
+      sysDescr_regex: '/^XOS/'
```
Comment (murrant) [inline includes/definitions/extremeware.yaml]: This should match no device right?  As you wrote it is looks for sysDescr that start with XOS.

Perhaps just
```
sysObjectId_regex: '/^.1.3.6.1.4.1.1916.2.1.([1-5]?[0-9]|6[0-5])$/'

```

**Comment by laf** [comment]
Updated

**Comment by murrant** [comment]
Go ahead and pick a day and send the notification.  Thanks.

**Comment by laf** [comment]
Not to be merged until the 10th.

**Comment by laf** [comment]
Thanks @murrant I didn't get much time last night :(

**Comment by murrant** [comment]
No worries.  I actually had some time :-)

**Comment by rasssta** [comment]
Please remove sensor warning. It need some refactoring to work properly. It's spamming++ new limits at the moment.

**Comment by murrant** [comment]
@rasssta Can you please report a new issue for this and supply the requested info.

**Pull request merged**